src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/CSharp/DefaultCSharpCodeActionProvider.cs

```diff
@@ -4,45 +4,42 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
-using System.Text.RegularExpressions;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.Models;
+using Microsoft.CodeAnalysis.ExternalAccess.Razor;
 using OmniSharp.Extensions.LanguageServer.Protocol.Models;
 
 namespace Microsoft.AspNetCore.Razor.LanguageServer.CodeActions
 {
     internal class DefaultCSharpCodeActionProvider : CSharpCodeActionProvider
     {
-        private static readonly HashSet<Regex> RegexMatchCodeActions = new HashSet<Regex>()
-        {
-            // Supports generating the empty constructor `ClassName()`, as well as constructor with args `ClassName(int)`
-            new Regex(@"^Generate constructor '.+\(.*\)'$", RegexOptions.IgnoreCase | RegexOptions.Compiled, TimeSpan.FromSeconds(1)),
-
-            new Regex("^Create and assign (property|field) '.+'$", RegexOptions.IgnoreCase | RegexOptions.Compiled, TimeSpan.FromSeconds(1))
-        };
-
-        private static readonly HashSet<string> StringMatchCodeActions = new HashSet<string>()
+        private static readonly HashSet<string> SupportedDefaultCodeActionNames = new HashSet<string>()
         {
+            // impl interface, abstract class,
+            // RazorPredefinedCodeRefactoringProviderNames.ImplementInterfaceExplicitly,
+            // Generate constructor '.+\(.*\)'
+            // Create and assign (property|field)
             "Generate Equals and GetHashCode",
```

Comment:
Once you can consume the ExternalAccess this will be replaced wit hthose types right?

Reply:
Yep.